### PR TITLE
docs: add user flow docs

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Adds some user flow documentation and explainers for the different modes. Feedback welcome :)

**Related Issues/PRs**
ref #11313

### CODE REVIEW

File: docs/user-flows.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+#### Benefits
+
+- Provides an overall performance score and all metrics.
+- Contains the most advice of all report types (both time-based and state-based audits are available).
+
+#### Limitations
+
+- Cannot analyze form submissions or single page app transitions.
+- Cannot analyze user interaction performance.
```
Comment (connorjclark) [inline docs/user-flows.md:29]:
```suggestion
- Cannot analyze user interactions.
```

File: docs/user-flows.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+```js
+import {writeFileSync} from 'fs';
+import puppeteer from 'puppeteer';
+import lighthouse from 'lighthouse';
+
+async function main() {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  const flow = new lighthouse.UserFlow(page);
```
Comment (connorjclark) [inline docs/user-flows.md:48]: nit: should we expose a `newUserFlow`? perhaps exposing the api via a class isn't necessary / ideal. idk

**Review by connorjclark — APPROVED** [review]
This looks great!

We should add a link to a sample flow report at the top of this page, for motivation.

File: docs/user-flows.md (hunk 2 above)
Comment (patrickhulce) [reply docs/user-flows.md:48]: cc @adamraine thoughts?

Comment (adamraine) [reply docs/user-flows.md:48]: The ES6 class made the most sense to me, is the benefit here that it matches the puppeteer methods?

Comment (patrickhulce) [reply docs/user-flows.md:48]: I think @connorjclark is referring to the *public* entrypoint being something non-constructor based, not that we should move away from the class itself.

I have to agree constructor-based public APIs feel gross to me for a node library, but admittedly I don't have a compelling case for why, so I'm fine with either.

Comment (adamraine) [reply docs/user-flows.md:48]: Aight I slept on this one and a function entry point SGTM.

Comment (patrickhulce) [reply docs/user-flows.md:48]: see #13139

File: docs/user-flows.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+async function main() {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  await page.goto('https://example.com');
+  const flow = await lighthouse.startFlow(page);
+
+  await flow.beginTimespan();
+  await page.type('#username', 'lighthouse');
+  await page.type('#password', 'LighthouseR0cks!');
```
Comment (adamraine) [inline docs/user-flows.md:94]: 🙃 
```suggestion
  await page.type('#password', 'Lighth0useR0cks!');
```

**Pull request merged**